modules/globebrowsing/tile/tileprovider/tileprovider.cpp

```diff
@@ -45,12 +45,13 @@ namespace tileprovider {
 unsigned int TileProvider::_numTileProviders = 0;
 
 std::unique_ptr<TileProvider> TileProvider::createFromDictionary(
+    layergroupid::TypeID layerTypeID,
     const ghoul::Dictionary& dictionary)
 {
-    std::string type = "LRUCaching";
-    dictionary.getValue(KeyType, type);
+    std::string type = layergroupid::LAYER_TYPE_NAMES[static_cast<int>(layerTypeID)];
     auto factory = FactoryManager::ref().factory<TileProvider>();
-    std::unique_ptr<TileProvider> result = factory->create(type, dictionary);
+    std::unique_ptr<TileProvider> result = nullptr;
```

Comment:
`std::unique_ptr<TileProvider> result = factory->create(type, dictionary);`
or directly returning the result